trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -259,12 +260,38 @@ def spin(self,conditions):
             F            = 2.*arccos_piece/pi
             Gamma        = vt*(4.*pi*r/B)*F*(1.+(4.*lamdaw*R/(pi*B*r))*(4.*lamdaw*R/(pi*B*r)))**0.5 
             Re           = (W*c)/nu  
-        
-            Cl    = np.zeros((ctrl_pts,Nr))              
-            Cdval = np.zeros((ctrl_pts,Nr)) 
-            for jj in range(Nr):                 
-                Cl[:,jj]    = cl_sur[a_geo[a_loc[jj]]](Re[:,jj],alpha[:,jj],grid=False)  
-                Cdval[:,jj] = cd_sur[a_geo[a_loc[jj]]](Re[:,jj],alpha[:,jj],grid=False) 
+            
+            # If propeller airfoils are defined, using airfoil surrogate 
+            if a_loc != None:
+                # Compute blade CL distribution from the airfoil data 
+                Cl    = np.zeros((ctrl_pts,Nr))              
+                Cdval = np.zeros((ctrl_pts,Nr)) 
+                for jj in range(Nr):                 
+                    Cl[:,jj]    = cl_sur[a_geo[a_loc[jj]]](Re[:,jj],alpha[:,jj],grid=False)  
```

Comment:
This loop needs to be  removed, it's going to be soooooooo sloooooww

Reply:
loop removed